wake-lock/wakelock-query.https.any.js

```diff
@@ -0,0 +1,144 @@
+  }
+}, "Filter all constructed WakeLock instances");
+
+test(() => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  // query "screen" type of WakeLock
+  const lockList1 = lock1.query({ type: "screen" });
+  assert_equals(lockList1.size, 2, "Size of WakeLock instance list with 'screen' type");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
+  }
+
+  // query "system" type of WakeLock
+  const lockList2 = lock1.query({ type: "system" });
+  assert_equals(lockList2.size, 2, "Size of WakeLock instance list with 'system' type");
+  for (let lock of lockList2) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "system", "WakeLock type");
+  }
+}, "Filter constructed WakeLock instances with 'type' option");
+
+promise_test(async t => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+
+  const controller = new AbortController();
+  await lock1.request({ signal: controller.signal });
+  assert_true(lock1.active, "lock1 must have been activated");
+  assert_true(lock2.active, "lock2 must have been activated");
+
+  // query activated WakeLock
+  const lockList1 = lock1.query({ active: true });
+  assert_equals(lockList1.size, 2, "Size of WakeLock instance list with activated state");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
+  }
+
+  // query inactivated WakeLock
+  const lockList2 = lock1.query({ active: false });
+  assert_equals(lockList2.size, 0, "Size of WakeLock instance list with activated state");
+
+  controller.abort();
+}, "Filter constructed WakeLock instances with 'active' option");
+
+promise_test(async t => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  const controller = new AbortController();
+  await lock1.request({ signal: controller.signal });
+  assert_true(lock1.active, "lock1 must have been activated");
+  assert_true(lock2.active, "lock2 must have been activated");
+  assert_false(lock3.active, "lock3 must have been activated");
+  assert_false(lock4.active, "lock4 must have been activated");
+
+  // query activated screen WakeLock
+  const lockList1 = lock1.query({ type: "screen", active: true });
+  assert_equals(lockList1.size, 2, "Size of activated screen WakeLock instance list");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
```

Comment:
```suggestion
    assert_equals(lock.type, "screen", "Got wrong WakeLock type");
```

Reply:
A bit concern here, the failure message from test run for `assert_equals` would look like:

_assert_equals: WakeLock type expected "screen" but got "xxx"_

That's why I only write "WakeLock type" as message.

Reply:
Ah! ok, good call.